examples/TPOT_comparison.ipynb

```diff
@@ -0,0 +1,304 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Setting the plotly template for the rest of the notebook to `plotly_white`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pio.templates.default = \"plotly_white\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load two datasets. As mentioned in the introduction, this analysis will only make sense if the 2 runs are compatible, that is the number of input tokens is the same, or close to."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result_1 = Result.load(\"<path of saved results of first run>\")\n",
+    "result_2 = Result.load(\"<path of saved results of second run>\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dimension = \"time_per_output_token\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## TPOT vs num of output tokens"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = scatter_histogram_2d(result_1, \"num_tokens_output\", dimension, 20, 20)\n",
+    "fig.update_layout(title=result_1.run_name)\n",
+    "fig"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = scatter_histogram_2d(result_2, \"num_tokens_output\", dimension, 20, 20)\n",
+    "fig.update_layout(title=result_2.run_name)\n",
+    "fig"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Distribution comparison\n",
+    "\n",
+    "It might be interesting to have a better understanding of the actual distribution of the TPOT, for example by observing the distribution using boxplots or histograms. We'll start by creating a boxplot for each run using , and then combining them to provide a clear comparison."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig = go.Figure()\n",
+    "\n",
+    "tr1 = boxplot_by_dimension(result=result_1, dimension=dimension)\n",
+    "tr2 = boxplot_by_dimension(result=result_2, dimension=dimension)\n",
+    "fig.add_traces(\n",
+    "    [\n",
+    "        tr1,\n",
+    "        tr2,\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "# use log scale for the time axis\n",
+    "fig.update_xaxes(type=\"log\")\n",
+    "\n",
+    "fig.update_layout(\n",
+    "    legend=dict(yanchor=\"top\", y=0.99, xanchor=\"right\", x=0.99),\n",
+    "    title=f\"Comparison of {dimension.replace('_', ' ').capitalize()}\",\n",
+    ")\n",
+    "fig"
```

Comment:
Can we include an example on how to override the legend here - even if it's just 'run 1' and 'run 2' by default? I tried loading `{model_id}/sweep/{timestamp}/00005-clients` to compare 2 models using the data generated by the 'Load testing' notebook, and the default legend on the comparison plot was just "00005-clients, 00005-clients" which was ofc unhelpful.

(Likewise for the TTFT comparison notebook)